packages/features/bookings/lib/getBookingResponsesSchema.ts

```diff
@@ -103,15 +102,41 @@ function preprocess<T extends z.ZodType>({
           };
           try {
             parsedValue = JSON.parse(value);
-          } catch (e) {}
+          } catch (e) {
+            console.error("Failed to parse JSON:", e);
+          }
           const optionsInputs = field.optionsInputs;
           const optionInputField = optionsInputs?.[parsedValue.value];
           if (optionInputField && optionInputField.type === "phone") {
-            parsedValue.optionValue = ensureValidPhoneNumber(parsedValue.optionValue);
+            let phoneValue = parsedValue.optionValue;
+            // Auto-prepend country code for single country restrictions
+            if (field.allowedCountryCodes?.length === 1 && phoneValue) {
+              const countryCode = field.allowedCountryCodes[0].toUpperCase();
+              const dialCode = getCountryCallingCode(countryCode as CountryCode);
+
+              // If phoneValue doesn't start with + or doesn't start with the correct country code
+              if (!phoneValue.startsWith(`+${dialCode}`)) {
+                // Remove any existing + and prepend the correct country code
+                phoneValue = `+${dialCode}${phoneValue.replace(/^\+/, "")}`;
+              }
+            }
+            parsedValue.optionValue = ensureValidPhoneNumber(phoneValue);
           }
           newResponses[field.name] = parsedValue;
         } else if (field.type === "phone") {
-          newResponses[field.name] = ensureValidPhoneNumber(value);
+          // Auto-prepend country code for single country restrictions
```

Comment:
I think the single country restriction causes issues for many countries. For example, in Austria if I don’t see +43, I naturally enter my number with a leading 0:
<img width="242" height="80" alt="Screenshot 2025-09-30 at 10 57 07 AM" src="https://github.com/user-attachments/assets/b6b83689-161c-4819-a2f3-696d6789ffd6" />

But then we change it to: +430680123456
<img width="273" height="85" alt="Screenshot 2025-09-30 at 11 00 01 AM" src="https://github.com/user-attachments/assets/366ce8ad-78c1-4026-917a-18b136a3aef1" />

Which is the wrong international format, as it should be +43 680123456